File: Python/Euklides.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  Euklides.py
#  
#  
#  
#  

def nwd_klasyczny(a, b):
    
    while a != b:
        if a > b:
            a = a - b
        else:
            b = b - a
    
    return a
    
def nww(a, b):
    
    
    
def main(args):
    
    a, b = 34, 65 
    print("NWD1({}, {}) = {}".format(a, b, nwd_klasyczny(a, b)))
    print("NWW({}, {}) = {}".format(a, b, nww(a, b)))
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))   

File: cpp/Euklides.cpp
/*
 * euklides.cpp
 */


#include <iostream>
using namespace std;
int nwd_1(int a, int b) {
    while(a != b) {
        if (a > b){
            a = a - b;
        }
        else {
            b = b - a;
        }
    }
    return(a);
}


int main(int argc, char **argv)
{
    int b = 0;
    int a = 0;
    cout << "Podaj 2 liczby: ";
    cin >> a >> b;

    int nwd = nwd_1(a,b);
        cout << "NWD ("<< a <<","<< b <<") = " << nwd << endl;

    return 0;
}
